pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -608,6 +599,17 @@ func (npm *nominatedPodMap) AddNominatedPod(pod *v1.Pod, nodeName string) {
 	npm.Unlock()
 }
 
+// NominatedPodsForNode returns pods that are nominated to run on the given node,
+// but they are waiting for other pods to be removed from the node before they
+// can be actually scheduled.
+func (npm *nominatedPodMap) NominatedPodsForNode(nodeName string) []*v1.Pod {
+	npm.RLock()
+	defer npm.RUnlock()
+	// TODO: we may need to return a copy of []*Pods to avoid modification
+	// on the caller side.
```

Comment:
when do we determine if the TODO is true or not? Seems like we'll probably never return if it doesn't cause a bug. Is there a way to add a unit test?

Reply:
Basically it's impossible to use UT to cover potential error is user side, esp. some users are out of k/k.

Still it's quite related with how caller use it - append pod to the slice is ok, but good with modifying the element using `slice[i]=newPod`.